File: Array/leetcode_26/leetcode_26.go
package main

import (
	"fmt"
)

func removeDuplicates(nums []int) int {
	var (
		n    = len(nums)
		left = 0
	)
	if n == 0 {
		return left
	}
	left++
	for right := 1; right < n; right++ {
		if nums[right] != nums[right-1] {
			nums[left] = nums[right]
			left++
		}
	}
	return left
}

func main() {
	nums := [...]int{1, 1, 2, 2}
	// nums := [...]int{0, 0, 1, 1, 1, 2, 2, 3, 3, 4}
	result := removeDuplicates(nums[:])
	fmt.Println(result)
}

File: Array/leetcode_26/leetcode_26.py
from typing import List

"""
26. 删除有序数组中的重复项
给你一个 升序排列 的数组 nums ，请你 原地 删除重复出现的元素，使每个元素 只出现一次 ，返回删除后数组的新长度。元素的 相对顺序 应该保持 一致 。然后返回 nums 中唯一元素的个数。
"""


class Solution1:
    def removeDuplicates(self, nums: List[int]) -> int:
        slow = 0
        fast = 0

        while fast < len(nums):
            if nums[slow] != nums[fast]:
                slow += 1
                nums[slow] = nums[fast]
            fast += 1

        return slow + 1


class Solution2:
    def removeDuplicates(self, nums: List[int]) -> int:
        slow = 0

        for fast in range(len(nums)):
            if nums[slow] != nums[fast]:
                slow += 1
                nums[slow] = nums[fast]

        return slow + 1
